docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -1613,7 +1613,7 @@ From, Join and Index by
     RangeVariableDeclaration                   ::= AbstractSchemaName ["AS"] AliasIdentificationVariable
     JoinAssociationDeclaration                 ::= JoinAssociationPathExpression ["AS"] AliasIdentificationVariable [IndexBy]
     Join                                       ::= ["LEFT" ["OUTER"] | "INNER"] "JOIN" (JoinAssociationDeclaration | RangeVariableDeclaration) ["WITH" ConditionalExpression]
-    IndexBy                                    ::= "INDEX" "BY" StateFieldPathExpression
```

Comment:
Please also add an example in the docs below after the language description in the `Using INDEX BY ` section.

Reply:
I've added a second example. Does it look okay?
I think the existing example is helpful for a more complex result with nested arrays and multiple levels of indexing, so I left that one there.